torch/csrc/utils/python_numbers.h

```diff
@@ -45,6 +65,14 @@ inline int64_t THPUtils_unpackLong(PyObject* obj) {
   return (int64_t)value;
 }
 
+inline uint32_t THPUtils_unpackUInt32(PyObject* obj) {
+  unsigned long value = PyLong_AsUnsignedLong(obj);
+  if (PyErr_Occurred()) {
+    throw python_error();
+  }
+  return (uint32_t)value;
```

Comment:
Actually, doesn't this introduce a potential error on Linux? In particular, if unsigned long happens to be 64 bits, then the Python conversion wouldn't overflow, but the `(uint32_t)` cast would overflow!

Reply:
You are right. I'll add a range check in it.

Reply:
Looks like this optimizes away, good show.